File: Bubble Sort/BubbleSortCSharp/BubbleSort.cs
namespace bubblesort
{
    public class bubblesort
    {
        public int[] BubbleSort(int[] array)
        {
            int tmp;

            for (int i = 0; i < array.Length; i++)
            {
                for (int y = i + 1; y < array.Length; y++)
                {
                    tmp = array[i];
                    if (tmp > array[y])
                    {
                        array[i] = array[y];
                        array[y] = tmp;
                    }
                }
            }
            return array;
        }
    }
}

File: Bubble Sort/BubbleSortRuby/BubbleSort.rb
#!/usr/bin/ruby
def bubble_sort(array)
   n = array.length
   swapped = true
   while swapped do
     swapped = false
     (n - 1).times do |i|
       if array[i] > array[i + 1]
         array[i], array[i + 1] = array[i + 1], array[i]
     swapped = true
       end
     end
   end
   array 
end

arr = [5,1,102,65,12,42,103,17]

puts arr
puts "Sorted"
puts bubble_sort(arr)
